src/chat/chat_panel.rs

```diff
@@ -929,6 +869,86 @@ impl ChatPanel {
             auto_scroll_cancellable: false,
         };
     }
+
+    fn update_visibilities(&mut self, scope: &mut Scope) {
+        macro_rules! show {
+            ($view_id:ident) => {
+                self.view(id!($view_id)).set_visible(true)
+            };
+        }
+
+        macro_rules! hide {
+            ($view_id:ident) => {
+                self.view(id!($view_id)).set_visible(false)
+            };
+        }
+
+        let store = scope.data.get_mut::<Store>().unwrap();
+
+        enum ComputedState {
```

Comment:
Technically, visibility can be modified directly inside the `if` conditions but declaring a list of all possible states makes the code a bit more declarative and easier to understand.